src/main/java/com/aizistral/nochatreports/mixins/both/MixinSerializer.java

```diff
@@ -0,0 +1,43 @@
+package com.aizistral.nochatreports.mixins.both;
+
+import com.aizistral.nochatreports.core.NoReportsConfig;
+import com.aizistral.nochatreports.core.NoReportServerStatus;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import net.minecraft.network.protocol.status.ServerStatus;
+import net.minecraft.util.GsonHelper;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import java.lang.reflect.Type;
+
+@Mixin(ServerStatus.Serializer.class)
+public class MixinSerializer {
+
+	@Inject(method = "serialize(Lnet/minecraft/network/protocol/status/ServerStatus;" +
+			"Ljava/lang/reflect/Type;Lcom/google/gson/JsonSerializationContext;)Lcom/google/gson/JsonElement;",
+			at = @At("RETURN"))
+	private void onSerialize(ServerStatus serverStatus, Type type, JsonSerializationContext context,
+			CallbackInfoReturnable<JsonElement> info) {
+		if (NoReportsConfig.includeQueryData() && serverStatus instanceof NoReportServerStatus noReportStatus) {
+			((JsonObject)info.getReturnValue()).addProperty("saveMinecraft", noReportStatus.hasChatReporting());
+		}
+	}
+
+	@Inject(method = "deserialize(Lcom/google/gson/JsonElement;Ljava/lang/reflect/Type;" +
+			"Lcom/google/gson/JsonDeserializationContext;)" +
+			"Lnet/minecraft/network/protocol/status/ServerStatus;",
+			locals = LocalCapture.CAPTURE_FAILSOFT, at = @At("RETURN"))
+	private void onDeserialize(JsonElement jsonElement, Type arg1, JsonDeserializationContext context,
+			CallbackInfoReturnable<ServerStatus> info, JsonObject jsonObj, ServerStatus status) {
+		if (jsonObj.has("saveMinecraft") && status instanceof NoReportServerStatus noReportStatus) {
```

Comment:
No you are correct, on the client this will always be false from the deserialization. Cause at first I wanted it to be both client and server although I didn't know if you wanted to implement it client-side so that you could add icons in the server list. Otherwise without any indicator its pretty useless on the client, although mods can still check if the json object has the `saveMinecraft` tag. I guess that does mean this Mixin is useless unless you implement the deserialize.

So I  guess it depends, do you want to use it on the client? If so, I can remove the instanceof check.

Reply:
I don't see a point to leave the implementation at server side only without doing anything with it on client.